Agent crash on Amazon Linux 2023 due to OpenSSL error

## Description

During testing of the agent MVP, the following failure occurred on Amazon Linux 2023 when connecting to the server that needs to be investigated and resolved:

```
× wazuh-agent.service - Wazuh agent
     Loaded: loaded (/usr/lib/systemd/system/wazuh-agent.service; disabled; preset: disabled)
     Active: failed (Result: core-dump) since Thu 2025-01-30 14:37:25 UTC; 12s ago
   Duration: 2.299s
    Process: 7101 ExecStart=/usr/bin/env /usr/share/wazuh-agent/bin/wazuh-agent (code=dumped, signal=SEGV)
   Main PID: 7101 (code=dumped, signal=SEGV)
        CPU: 53ms

Jan 30 14:37:23 ip-172-31-67-15.ec2.internal systemd[1]: Started wazuh-agent.service - Wazuh agent.
Jan 30 14:37:23 ip-172-31-67-15.ec2.internal env[7101]: [2025-01-30 14:37:23.298] [wazuh-agent] [error] [ERROR] [http_client.cpp:301] [PerformHttpRequest] Error: context: unknown option (digital envelope routines) [asio.ssl:50331817]
Jan 30 14:37:23 ip-172-31-67-15.ec2.internal env[7101]: [2025-01-30 14:37:23.298] [wazuh-agent] [warning] [WARN] [http_client.cpp:353] [AuthenticateWithUuidAndKey] Error: 500.
Jan 30 14:37:23 ip-172-31-67-15.ec2.internal env[7101]: [2025-01-30 14:37:23.298] [wazuh-agent] [warning] [WARN] [communicator.cpp:99] [SendAuthenticationRequest] Failed to authenticate with the manager. Retrying in 30 seconds.
Jan 30 14:37:25 ip-172-31-67-15.ec2.internal systemd[1]: wazuh-agent.service: Main process exited, code=dumped, status=11/SEGV
Jan 30 14:37:25 ip-172-31-67-15.ec2.internal systemd[1]: wazuh-agent.service: Failed with result 'core-dump'.
```

Maybe related to: https://github.com/openssl/openssl/discussions/23016

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -27,6 +27,8 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux" OR CMAKE_SYSTEM_NAME STREQUAL "Darwin")
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -static-libstdc++")
 endif()
 
+find_package(OpenSSL REQUIRED)
+
 include(cmake/CommonSettings.cmake)
 set_common_settings()
 
@@ -46,7 +48,7 @@ add_subdirectory(modules)
 add_subdirectory(agent)
 
 add_executable(wazuh-agent agent/src/main.cpp)
-target_link_libraries(wazuh-agent Agent Logger)
+target_link_libraries(wazuh-agent PRIVATE Agent Logger OpenSSL::SSL OpenSSL::Crypto)
 
 if(WIN32)
     include(cmake/SetWindowsManifest.cmake)

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,5 +1,7 @@
 cmake_minimum_required(VERSION 3.22)
 
+set(OPENSSL_NO_AUTOLOAD_CONFIG ON)
+
 if(DEFINED ENV{VCPKG_ROOT})
     message(STATUS "VCPKG is installed. VCPKG_ROOT: $ENV{VCPKG_ROOT}")
     set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake" CACHE FILEPATH "")

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,7 +1,5 @@
 cmake_minimum_required(VERSION 3.22)
 
-set(OPENSSL_NO_AUTOLOAD_CONFIG ON)
-
 if(DEFINED ENV{VCPKG_ROOT})
     message(STATUS "VCPKG is installed. VCPKG_ROOT: $ENV{VCPKG_ROOT}")
     set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake" CACHE FILEPATH "")

--- a/src/agent/src/main.cpp
+++ b/src/agent/src/main.cpp
@@ -28,9 +28,17 @@ static const auto OPT_REMOVE_SERVICE {"remove-service"};
 static const auto OPT_RUN_SERVICE {"run-service"};
 #endif
 
+#include <openssl/provider.h>
+
 int main(int argc, char* argv[])
 {
     Logger logger;
+
+    auto legacyProvider = OSSL_PROVIDER_load(nullptr, "legacy");
+    if (!legacyProvider)
+    {
+        LogWarn("Failed to load the OpenSSL legacy provider.");
+    }
 
     try
     {

--- a/src/agent/src/main.cpp
+++ b/src/agent/src/main.cpp
@@ -28,17 +28,15 @@ static const auto OPT_REMOVE_SERVICE {"remove-service"};
 static const auto OPT_RUN_SERVICE {"run-service"};
 #endif
 
-#include <openssl/provider.h>
+#include <openssl/ssl.h>
 
 int main(int argc, char* argv[])
 {
     Logger logger;
 
-    auto legacyProvider = OSSL_PROVIDER_load(nullptr, "legacy");
-    if (!legacyProvider)
-    {
-        LogWarn("Failed to load the OpenSSL legacy provider.");
-    }
+    SSL_library_init();
+    SSL_load_error_strings();
+    OpenSSL_add_all_algorithms();
 
     try
     {

